frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -333,6 +340,75 @@ pub mod pallet {
+        ///
+        /// No unclaimed rewards must remain prior to this call (otherwise it will fail).
+        /// Make sure to claim all the contract rewards prior to unregistering it.
+        /// TODO: weight
+        #[pallet::weight(1_000_000_000)]
+        pub fn unregister(
+            origin: OriginFor<T>,
+            contract_id: T::SmartContract,
+        ) -> DispatchResultWithPostInfo {
+            let developer = ensure_signed(origin)?;
+
+            let registered_contract =
+                RegisteredDevelopers::<T>::get(&developer).ok_or(Error::<T>::NotOwnedContract)?;
+
+            // This is a sanity check for the unregistration since it requires the caller
+            // to input the correct contract address.
+            ensure!(
+                registered_contract == contract_id,
+                Error::<T>::NotOwnedContract
+            );
+
+            let current_era = Self::current_era();
+            // in case contract wasn't staked yet
+            let last_claim_era =
+                Self::contract_last_claimed(&contract_id).unwrap_or(current_era.clone());
+
+            // Ensure that all contract rewards have been claimed prior to unregistration.
+            ensure!(
+                last_claim_era == current_era,
+                Error::<T>::ContractRewardsNotClaimed
+            );
+
+            // We need to unstake all funds that are currently staked
+            let era_staking_points =
+                ContractEraStake::<T>::take(&contract_id, &current_era).unwrap_or_default();
+            let mut unstake_accumulator = BalanceOf::<T>::zero();
+            for (staker_id, amount) in era_staking_points.stakers.iter() {
+                let mut ledger = Self::ledger(staker_id);
+                ledger = ledger.saturating_sub(*amount);
+                Self::update_ledger(staker_id, &ledger);
+
+                // increment total unstake value accumulator
+                unstake_accumulator += *amount;
+            }
+
+            // Need to update total amount staked
+            let mut reward_and_stake =
+                Self::era_reward_and_stake(&current_era).ok_or(Error::<T>::UnexpectedState)?;
+            reward_and_stake.staked = reward_and_stake.staked.saturating_sub(unstake_accumulator);
+            EraRewardsAndStakes::<T>::insert(&current_era, reward_and_stake);
+
+            // Continue with cleanup
+            T::Currency::unreserve(&developer, T::RegisterDeposit::get());
+            RegisteredDapps::<T>::remove(&contract_id);
+            RegisteredDevelopers::<T>::remove(&developer);
+            ContractLastStaked::<T>::remove(&contract_id);
+            ContractLastClaimed::<T>::remove(&contract_id);
+
+            // TODO: Should this be removed? We lose all the information about the rewards earned by all stakers of this contract.
+            RewardsClaimed::<T>::remove_prefix(&contract_id, None);
```

Comment:
We do not expect this functionality to be used a lot and I would suggest to call remove_prefix()
Anyways there are other storage data which the frontend is using that will not be available after contract is removed. Even if  RewardsClaimed is left (not cleared), the frontend needs to remove this contract from it's database and stakers will not be able to check RewardsClaimed.
Just remove TODO comment

Reply:
All of what you wrote makes sense.

I was thinking more in line of UX where user might want to keep this as history.
But I guess we can store that data into DB.